runtime/stream.c

```diff
@@ -480,7 +480,13 @@ strmNextFile(strm_t *pThis)
 	 * make sure their iMaxFiles is large enough. But it is well-desired for other
 	 * use cases, e.g. a circular output log file. -- rgerhards, 2008-01-10
 	 */
-	pThis->iCurrFNum = (pThis->iCurrFNum + 1) % pThis->iMaxFiles;
+	if((pThis->iCurrFNum + 1) == 0 || pThis->iMaxFiles == 0) {
```

Comment:
I think the question can also be reworded. looking at the assertion, iMaxfiles should never be zero. So I think the question really is why this invariant does not hold. Obviously there is a problem, but we should possibly try to identify the root cause.